### PR TITLE
fix: `CompilerHost.getSourceFile` is being called for odd filenames

### PR DESCRIPTION
**Pull request by alan-agius4**
Would love to add some Unit Tests but kinda can't find a way where to test it. If any one is willing to give me pointers on how do you guy usually test this. I'll happily add them

Ignore falsy file names from `getSourceFileFromReferenceWorker`

Closes: #13629

### CODE REVIEW

**Comment by mhegazy** [comment]
@alan-agius4 we will need a unit test to go with this change. i would say you need a new file, since we do not have any existing unit test files that cover this area.
you can look at https://github.com/Microsoft/TypeScript/blob/master/src/harness/unittests/hostNewLineSupport.ts for reference for testing compilerHost use.

**Comment by alan-agius4** [comment]
@mhegazy I added the some tests.

**Comment by alan-agius4** [comment]
@mhegazy, thanks for the example, that was helpful.

I added the some tests.

File: src/compiler/program.ts [hunk 1]
```diff
@@ -1703,7 +1703,11 @@ namespace ts {
             fail?: (diagnostic: DiagnosticMessage, ...argument: string[]) => void,
             refFile?: SourceFile): SourceFile | undefined {
 
-            if (hasExtension(fileName)) {
+            // Don't process falsy fileNames
```
Comment (mhegazy) [inline src/compiler/program.ts]: how about doing this check on the call to `getDefaultLibraryFileName` instead. it is making me a bit nervous to have it in this general utility function, just to handle the return of `getDefaultLibraryFileName`.

Comment (alan-agius4) [reply src/compiler/program.ts]: Making you nervous is not a good sign.

You mean here: https://github.com/alan-agius4/TypeScript/blob/a27cd8f9a2e83d4b6fbf4e77c5041d94b01d86f6/src/compiler/program.ts#L611

Change it to: `if (!skipDefaultLib && defaultLibraryPath) `

**Comment by mhegazy** [comment]
thanks. this seems much targeted now. sorry for the back and forth.

**Pull request merged**